lib/internal/util/inspect.js

```diff
@@ -1458,15 +1458,11 @@ function handleMaxCallStackSize(ctx, err, constructorName, indentationLvl) {
 }
 
 function addNumericSeparator(integerString) {
-  let result = '';
-  let i = integerString.length;
-  const start = integerString.startsWith('-') ? 1 : 0;
-  for (; i >= start + 4; i -= 3) {
-    result = `_${integerString.slice(i - 3, i)}${result}`;
-  }
-  return i === integerString.length ?
-    integerString :
-    `${integerString.slice(0, i)}${result}`;
+  return StringPrototypeReplace(
+    integerString,
+    /[0-9](?=(?:[0-9]{3})+(?![0-9]))/g,
```

Comment:
`%String.prototype.replace%` calls the `@@replace` method of the regex, which defeats the purpose of primordials. 

<details>

```js
// User-land
RegExp.prototype[Symbol.replace] = () => 'foo';
String.prototype[Symbol.replace] = () => 'baz';

// Core
console.log(StringPrototypeReplace('baar', /a/, 'e')); // 'foo'
console.log(StringPrototypeReplace('baar', 'a', 'e')); // 'baz'
console.log(RegExpPrototypeSymbolReplaceAll('baar', 'a', 'e')); // 'baz'
console.log(RegExpPrototypeSymbolReplace(/a/, 'baar', 'e')); // 'bear'
console.log(RegExpPrototypeSymbolReplace(/a/g, 'baar', 'e')); // 'beer'
```

</details>

We should call `RegExpPrototypeSymbolReplace` directly here:
```suggestion
  return RegExpPrototypeSymbolReplace(
    /[0-9](?=(?:[0-9]{3})+(?![0-9]))/g,
    integerString,
```

Reply:
aha, thanks :-)